arangopipe/arangopipe/arangopipe_storage/arangopipe_admin_api.py

```diff
@@ -40,287 +29,73 @@
-
-        # if self.mscp.DB_ROOT_USER_PASSWORD in self.cfg["arangodb"]:
-        #     logger.info("A root user password was specified, persisting...")
-
-        #Provision the graph
-        self.create_enterprise_ml_graph(db_replication_factor)
-
-        # try:
-        #     self.create_db(
-        #         db_serv_host,
-        #         db_serv_port,
-        #         db_serv_name,
-        #         db_end_point,
-        #         db_dbName,
-        #         db_user_name,
-        #         db_password,
-        #         db_conn_protocol,
-        #     )
-
-        #     # If you could create a DB, proceed with provisioning the graph.
-        #     # Otherwise you had an issue creating the database.
-        #     if self.db is not None:
-        #         self.create_enterprise_ml_graph(db_replication_factor)
-
-        #         if persist_conn:
-        #             self.config.dump_data()
-        # except Exception:
-        #     logger.error("Error connecting to DB, trying again...")
-        #     time.sleep(2)
-        #     self.create_db(
-        #         db_serv_host,
-        #         db_serv_port,
-        #         db_serv_name,
-        #         db_end_point,
-        #         db_dbName,
-        #         db_user_name,
-        #         db_password,
-        #         db_conn_protocol,
-        #     )
-        #     self.create_enterprise_ml_graph(db_replication_factor)
-        #     if persist_conn:
-        #         self.config.dump_data()
-        #     logger.error("Obtained connection with retry!")
-
-        return
-
-    def check_repeated_creation(self, api_data):
-        if not api_data:
-            repeated_connection = False
-        else:
-            try:
-                user_name_equal = (api_data[self.mscp.DB_USER_NAME] == self.
-                                   cfg["arangodb"][self.mscp.DB_USER_NAME])
-                password_equal = (api_data[self.mscp.DB_PASSWORD] ==
-                                  self.cfg["arangodb"][self.mscp.DB_PASSWORD])
-                db_name_equal = (api_data[self.mscp.DB_NAME] ==
-                                 self.cfg["arangodb"][self.mscp.DB_NAME])
-                repeated_connection = user_name_equal or password_equal or db_name_equal
-                if user_name_equal:
-                    logger.info(
-                        "Attempting to create a connection with an existing username"
-                    )
-                if db_name_equal:
-                    logger.info(
-                        "Attempting to create a connection with an existing db name"
-                    )
-            except KeyError:
-                repeated_connection = False
+        self.graph_name = graph_name
+        self.db_replication_factor = replication_factor
+        self.create_graph = create_graph
+        self.reuse_connection = reuse_connection
 
-        return repeated_connection
+        # Provision the graph
+        if self.create_graph:
```

Comment:
@tjoubert @cw00dw0rd didn't we want to move this to the `ArangoPipe` class in `arangopipe_api.py`?

Reply:
Done. Do we also remove all graph related methods like `remove_edge_definition_from_arangopipe()`, `add_edges_to_arangopipe`, etc...?